sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/BufferedHttpResponse.java

```diff
@@ -53,12 +57,12 @@ public HttpHeaders getHeaders() {
 
     @Override
     public Flux<ByteBuffer> getBody() {
-        return cachedBody;
+        return cachedBody.flatMapMany(Flux::fromIterable).map(ByteBuffer::duplicate);
     }
 
     @Override
     public Mono<byte[]> getBodyAsByteArray() {
-        return cachedBody.next().map(ByteBuffer::array);
+        return FluxUtil.collectBytesInByteBufferStream(getBody(), (int) cachedBodySize.get());
```

Comment:
If the cachedBodySize is greater than Integer.MAX_VALUE then will this return a truncated byte[]? We should instead add the following check and throw:
```java
if(cachedBodySize.get() > Integer.MAX_VALUE ) {
     return monoError(logger, new IllegalStateException("Response body size doesn't fit into a byte array"));
}
```

Reply:
That is a good addition, should also see if it would be possible to add this into the other `HttpResponse` implementations to prevent needless array creations when we know it'll fail.